### DISCUSSION TITLE
Can I use a separate I2C bus with U2G8 TC Menus?

### DISCUSSION THREAD

**Discussion by leelx78** [davetcc/tcMenu#429] [Q&A] [answered]
Hi my I2C bus is a bit busy so I'm thinking to have the LCS (SSD1306 I2C) on a separate Wire/TwoWire object.

I've been looking at the source and trying things around but I'm struggling to find the proper way to do this without hardcoding anything in the automatically generated TCMenus files or in the libraries. 

What is the proper way to this? I see that tcmenuU8g2.h has
`    explicit U8g2Drawable(U8G2* u8g2, TwoWire* wire = nullptr);`

Is that the way to connect an external TwoWire object?

Thanks a lot!

**Comment by leelx78** [comment]
ok I managed to move everything else to Wire1 and it seems to work a lot better. Would be still good to have an "easy" way to select which I2C Bus should be connected to the screen if possible.

**Reply by davetcc** [reply]
apologies for the dealy in getting back, the quick starts don't support that at the moment. I'll look at how much work it would be to add it. It would need some thought as that plugin is already a bit complex. Further, it would be completely incompatible with the "reduce latency option".

**Reply by davetcc** [reply, marked as answer]
Although the Custom U8G2 plugin allows you to define your own U8G2 variable and just reference it in the plugin. That is the way to go for more complex cases. In that case all you need to define in your code is the actual display variable, and refer to that where the plugin asks for the variable name.
